conflagration/namespace.py

```diff
@@ -52,7 +54,10 @@ class LowerCaseKeys(_BaseNamespaceModifier):
     def __call__(self, original_dict):
         for k in original_dict.keys():
             if k.lower() != k:
-                original_dict[k.lower()] = original_dict[k]
+                if k.lower() in original_dict:
+                    original_dict[k.lower()].update(original_dict[k])
```

Comment:
This might not work, since you're updating the dict while looping through it's keys.  I think it's ok in earlier versions of python, but possibly not in 3 and forward.